source/features/link-to-file-in-file-history.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+This feature adds direct link to file/directory when viewing the history.
+GitHub provides a link but it points to the root of the repo which is not that helpful.
+
+See it in action at https://github.com/sindresorhus/refined-github/commits/master/readme.md
+*/
+
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import {getOwnerAndRepo} from '../libs/utils';
+import {file} from '../libs/icons';
+import {groupButtons} from '../libs/group-buttons';
+
+function init(): void {
+	const {repoName} = getOwnerAndRepo();
+
+	const pathToFile = select('.breadcrumb')!.textContent!.trim().replace(new RegExp(`^History for ${repoName}`), '');
+
+	for (const commitLinkCell of select.all<HTMLDivElement>('.commit-links-cell')) {
```

Comment:
What if you select the `tree` link that’s going to be next to this button? And then use `.closest` to get this cell. I think this enables some optimizations

Reply:
What kind of optimizations? Can you please tell me more? I selected the parent first and then applied the `tree` link selectors to reduces the chances of any false matches.

Reply:
fe9629b

I always try to reuse elements as much as possible, for example in this case I just got one element to get:

- insertion position
- file path (which it turns out isn't needed)
- starting point to add `width: auto`

Reply:
Simply great! 👍 I wasn't aware of this automatic redirection thing. 😄 TIL :)